binaries/daemon/src/spawn.rs

```diff
@@ -301,14 +301,44 @@ pub async fn spawn_node(
                     cmd
                 }
             } else if python_operators.is_empty() && other_operators {
-                let mut cmd = tokio::process::Command::new(
-                    std::env::current_exe().wrap_err("failed to get current executable path")?,
-                );
-                cmd.arg("runtime");
-                cmd
+                let current_exe =
+                    std::env::current_exe().wrap_err("failed to get current executable path")?;
+                let mut file_name = current_exe.clone();
+                file_name.set_extension("");
+                let file_name = file_name
+                    .file_name()
+                    .and_then(|s| s.to_str())
+                    .context("failed to get file name from current executable")?;
+
+                // Check if the current executable is a python binary meaning that dora is installed within the python environment
+                if file_name.ends_with("python") || file_name.ends_with("python3") {
+                    // Use the current executable to spawn runtime
+                    let python = get_python_path()
+                        .wrap_err("Could not find python path when spawning custom node")?;
+                    let mut cmd = tokio::process::Command::new(python);
+
+                    tracing::info!(
+                        "spawning: python -uc import dora; dora.start_runtime() # {}",
+                        node.id
+                    );
+
+                    cmd.args([
+                        "-c",
+                        format!("import dora; dora.start_runtime() # {}", node.id).as_str(),
+                    ]);
+                    cmd
+                } else {
+                    let mut cmd = tokio::process::Command::new(
+                        std::env::current_exe()
+                            .wrap_err("failed to get current executable path")?,
+                    );
+                    cmd.arg("runtime");
+                    cmd
+                }
             } else {
-                eyre::bail!("Runtime can not mix Python Operator with other type of operator.");
+                bail!("Could not figure out dora installation. Could you try to reinstall dora or run it with `dora` command?");
```

Comment:
This `else` branch is entered when there are both python and non-python operators for the same runtime node. So this new error message is not correct.